src/Simplex_tree/include/gudhi/Simplex_tree.h

```diff
@@ -165,6 +173,22 @@ class Simplex_tree {
     }
   };
 
+ private:
+  /** \brief An iterator for an optimized search for the star of a simplex.
+   *
+   * \details It requires the Options::link_nodes_by_label to be true and store two
+   * extra pointers in each node of the simplex tree. The Nodes of same label are
+   * linked in a list.
+   */
+  using Optimized_star_simplex_iterator = Simplex_tree_optimized_star_simplex_iterator<Simplex_tree>;
+  /** \brief Range for an optimized search for the star of a simplex. */
+  using Optimized_star_simplex_range = boost::iterator_range<Optimized_star_simplex_iterator>;
+
+  using Const_boost_iterator = const boost::container::vec_iterator<std::pair<int, Node >*, false>;
+  using Filtered_boost_predicate = std::function<bool(Const_boost_iterator)>;
+  std::function<bool(Const_boost_iterator)> select;
```

Comment:
select is a class member because `boost::adaptors::filter` returns a `boost::filtered_range<decltype(select), Optimized_star_simplex_range>;`.

I opened an [issue on boost range documentation](https://github.com/boostorg/range/pull/147) because this was missing.

I am also not a big fan of `std::function`

Reply:
I don't understand how that's related. `decltype(select)` is just `Filtered_boost_predicate`.

Reply:
Ok, now I got it. I moved select from class attribute to a variable in `cofaces_simplex_range` method on e9f7c6a

Reply:
I replaced `std::function` with a functor on [8c21fdf](https://github.com/GUDHI/gudhi-devel/pull/877/commits/8c21fdf5d37cb65da8e42f5dba64b27ef3218307). Timings are almost the same.